File: Data Structures & Algorithms/concatenation-of-array/submission-3.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> sol;
        for(int i = 0; i < 2; i++) {
            for(int num : nums) {
                sol.push_back(num);
            }

        }
        return sol;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:

        answer = nums[0]
        l = 0
        r = len(nums) - 1

        while l <= r:
            m = (l+r) // 2

            if nums[l] < nums[r]:
                answer = min(answer, nums[l])
                break

            

            answer = min(answer, nums[m])

            if nums[m] >= nums[l]:
                l = m + 1
            else:
                r = m - 1



        return answer
        
        